### PR TITLE
ADD pull request protocol into the contribution guidelines document

### PR DESCRIPTION
**Pull request by fgalan**

### CODE REVIEW

File: doc/manuals/contribution_guidelines.md [hunk 1]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
 Note that contribution workflows themselves (e.g. pull requests, etc.) are described in another document 
 ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)).
 
+## Pull Request protocol
+
+As explained in ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)) 
+contributions are done using pull request (PR). The detailed "protocol" used in such PRs is described below:
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: a pull request

File: doc/manuals/contribution_guidelines.md [hunk 2]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+
+* Direct commits to master branch (even single-line modifications) are not allowed. Every modification has to come as a PR
+* In the case the PR is implemeting/fixing a numbered issue, the issue number has to be referenced in the body of the PR at creating time
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: implementing

at creation time

File: doc/manuals/contribution_guidelines.md [hunk 3]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+* Anybody is welcome to provide comments in the PR (either direct comments or using the review github feature)
+* Use *code line comments* instead of *general comments*, for traceability reasons (see comments lifecycle below)
+* Comments lifecycle
+  * Comment is created, initiating a *comment thread*
+  * New comments can be added as responses to the original one, starting a discussion
+  * After discussion, the comment thread ends in one of the following ways
+    * `Fix in <commit hash>` in the case the discussion involves a fix in the PR branch (which commit hash is
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: Fixed

File: doc/manuals/contribution_guidelines.md (hunk 1 above)
Comment (jmcanterafonseca) [reply doc/manuals/contribution_guidelines.md]: PR

File: doc/manuals/contribution_guidelines.md (hunk 2 above)
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: In case

File: doc/manuals/contribution_guidelines.md [hunk 4]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
 ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)).
 
+## Pull Request protocol
+
+As explained in ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)) 
+contributions are done using pull request (PR). The detailed "protocol" used in such PRs is described below:
+
+* Direct commits to master branch (even single-line modifications) are not allowed. Every modification has to come as a PR
+* In the case the PR is implemeting/fixing a numbered issue, the issue number has to be referenced in the body of the PR at creating time
+* Anybody is welcome to provide comments in the PR (either direct comments or using the review github feature)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: using the review feature offered by Github

File: doc/manuals/contribution_guidelines.md [hunk 5]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+* Use *code line comments* instead of *general comments*, for traceability reasons (see comments lifecycle below)
+* Comments lifecycle
+  * Comment is created, initiating a *comment thread*
+  * New comments can be added as responses to the original one, starting a discussion
+  * After discussion, the comment thread ends in one of the following ways
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: ways:

File: doc/manuals/contribution_guidelines.md [hunk 6]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+    * `Fix in <commit hash>` in the case the discussion involves a fix in the PR branch (which commit hash is 
+       included as reference)
+    * `NTC`, in the case at the end nothing needs to be done (NTC = Nothing To Change)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: if finally nothing needs to be done

File: doc/manuals/contribution_guidelines.md [hunk 7]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+ * PR can be merged when all the following conditions are met
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: "all" is needed?

met:

File: doc/manuals/contribution_guidelines.md [hunk 8]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+    * All comment threads are closed
+    * Anybody who has taken part in some discussion provides a `LGTM` general comment (LGTM = Looks good to me)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: All the participants in the discussion have provided

File: doc/manuals/contribution_guidelines.md [hunk 9]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+ * Self-merging is not allowed (except in rare and justified circustances)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: circumstances and previous approval of the project owners and peers

File: doc/manuals/contribution_guidelines.md (hunk 1 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 2 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 4 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 5 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 3 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 6 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 7 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 8 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md (hunk 9 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 080e9848ee2d867a54bbd735591f90f3c2948dfa

File: doc/manuals/contribution_guidelines.md [hunk 10]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
 ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)).
 
+## Pull Request protocol
+
+As explained in ([FIWARE Development Guidelines](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Developer_Guidelines)) 
+contributions are done using a pull request (PR). The detailed "protocol" used in such PR is described below:
+
+* Direct commits to master branch (even single-line modifications) are not allowed. Every modification has to come as a PR
+* In case the PR is implementing/fixing a numbered issue, the issue number has to be referenced in the body of the PR at creation time
+* Anybody is welcome to provide comments in the PR (either direct comments or using the review feature offered by Github)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: to the PR

File: doc/manuals/contribution_guidelines.md [hunk 11]
```diff
@@ -19,6 +19,27 @@ planning to contribute to the code you should read this document and get familia
...
+  * Comment is created, initiating a *comment thread*
+  * New comments can be added as responses to the original one, starting a discussion
+  * After discussion, the comment thread ends in one of the following ways:
+    * `Fixed in <commit hash>` in case the discussion involves a fix in the PR branch (which commit hash is 
+       included as reference)
+    * `NTC`, if finally nothing needs to be done (NTC = Nothing To Change)
+ * PR can be merged when the following conditions are met:
+    * All comment threads are closed
+    * All the participants in the discussion have provided a `LGTM` general comment (LGTM = Looks good to me)
+ * Self-merging is not allowed (except in rare and justified circustances)
```
Comment (jmcanterafonseca) [inline doc/manuals/contribution_guidelines.md]: still needs to be changed to circumstances

**Comment by jmcanterafonseca** [comment]
LGTM with the latest two comments fixed

File: doc/manuals/contribution_guidelines.md (hunk 10 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 83df10cc1acd354fce084e7f7e40f3d5f1fd14ad

File: doc/manuals/contribution_guidelines.md (hunk 11 above)
Comment (fgalan) [reply doc/manuals/contribution_guidelines.md]: Fixed in 83df10cc1acd354fce084e7f7e40f3d5f1fd14ad

**Pull request merged**